src/diffusers/loaders/unet.py

```diff
@@ -802,13 +805,20 @@ def _convert_ip_adapter_attn_to_diffusers(self, state_dicts, low_cpu_mem_usage=F
                 hidden_size = self.config.block_out_channels[block_id]
 
             if cross_attention_dim is None or "motion_modules" in name:
-                attn_processor_class = self.attn_processors[name].__class__
-                attn_procs[name] = attn_processor_class()
-
-            else:
-                attn_processor_class = (
-                    IPAdapterAttnProcessor2_0 if hasattr(F, "scaled_dot_product_attention") else IPAdapterAttnProcessor
-                )
+                if ('XFormers' not in str(self.attn_processors[name].__class__)):                
+                    attn_processor_class = (
+                        AttnProcessor2_0 if hasattr(F, "scaled_dot_product_attention") else AttnProcessor
+                    )
+                    attn_procs[name] = attn_processor_class()
+                else:
+                    attn_procs[name] = self.attn_processors[name]
+            else:                     
```

Comment:
can you explain the change here?

Reply:
Here in Else: it seems to me a situation of which comes first, the chicken or the egg? When we do not use the pipe.enable_xformers_memory_efficient_attention() call by default it is defining IPAdapterAttnProcessor2_0 or IPAdapterAttnProcessor for the IP Adapter. So when you call pipe.enable_xformers_memory_efficient_attention() before loading the IP Adapter all attn are defined for XFormersAttnProcessor, so when loading the IP Adapter modules after this call it is necessary to check if the defined mechanism is xformers to apply the new class "IPAdapterXFormersAttnProcessor". However, when you call pipe.enable_xformers_memory_efficient_attention() after loading the IP Adapter modules, the modules had already been set by default to "IPAdapterAttnProcessor2_0 or IPAdapterAttnProcessor " and the "set_use_memory_efficient_attention_xformers" method of the "Attention" class only knows how to set everything to XFormersAttnProcessor and this generated the error that was reported in the open issue. Now with the implementation that I made in this class, the method also knows how to identify "IPAdapterAttnProcessor2_0 or IPAdapterAttnProcessor " in the modules and correctly replace them with the new class. But it only knows how to do this because "IPAdapterAttnProcessor2_0 or IPAdapterAttnProcessor " was defined when loading the module. So these checks are necessary on both sides due to the order in which pipe.enable_xformers_memory_efficient_attention() is called, before or after loading the modules.

Reply:
Can you confirm that you added this change in order to be able to handle this?

```pytnon
pipe.enable_xformers_memory_efficient_attention()
pipe.load_ip_adapter()
```

Reply:
Yes this is to solve this order, and vice-versa. My provided code in this PR simulate the two scenarios